### DISCUSSION TITLE
Best general way to test based on multilayer informations if two structures are equal?

### DISCUSSION THREAD

**Discussion by simonmb** [rdkit/rdkit#4580] [Q&A] [answered]
I am looking for a general way to check whether two molecules are the same? The best would be if I could also specify whether to ignore stereochemistry information or not.

I found this old discussion: https://sourceforge.net/p/rdkit/mailman/rdkit-discuss/thread/011CC9451532CC45BD0F5EC53EADCA7E1C7A3FDB%40UKDXMLLYC305.rf.lilly.com/

where @greglandrum recommends to standardize the structure with CheckMolecule from the AvalonTools and then compare the different parts of the InCHIs depending on what layer of information one is interested in.

I had been using a Tanimoto similarity of "1" for the Morgan fingerprint with a radius of 2 to check whether two molecules are the same. So far that worked, at least for the examples I reviewed by hand. But I was wondering if I have been getting false positives with this comparison method.

**Comment by ptosco** [comment, marked as answer]
You can use the `MolStandardize.rdMolStandardize` functionality ahead of your comparison to standardize the two molecules wrt functional group notation, protonation state, counterion (`Normalizer`, `Reionizer`,  `LargestFragmentChooser`, `Uncharger`. optionally `MetalDisconnector`).
For stereochemistry, `Chem.MolToSmiles` has a flag to choose whether to keep stereochemistry or not in the output SMILES, and can be conveniently used to strip stereochemistry information from a molecule.
For tautomerism, you may either leverage InChI layers (as you suggest) or compare the two molecules' `MolHash.HetAtomTautomer` hashes which are invariant on the most common tautomeric states (e.g., 1,3-prototropy).

**Reply by simonmb** [reply]
Could you maybe give some advice as to what of the functions I actually have to apply?

I mainly have two groups of molecules I need to compare:
- neutral organic components
- charged species where I know the correct formal charge of.

What standardization would you recommend for each of these types? With the second type I have had problems because some sources position the charge on different atoms.

**Comment by greglandrum** [comment]
One additional detail for Paolo's response: a fingerprint similarity of 1.0 definitely does not mean that two molecules are identical: it's entirely possible for differing molecules to have identical fingerprints
